src/main/java/io/katharsis/dispatcher/JsonApiDispatcherImpl.java

```
@@ -0,0 +1,97 @@
+
+    private final JsonApiGet apiGet;
+    private final JsonApiPost apiPost;
+    private final JsonApiPatch apiPatch;
+    private final JsonApiDelete apiDelete;
+    private ExceptionMapperRegistry exceptionMapperRegistry;
+
+    /**
+     * Dispatch the request from a client
+     *
+     * @param request - the request we need to process
+     * @return the response form the Katharsis
+     */
+    @Override
+    public ResponseContext handle(Request request) {
+        ResponseContext response = null;
+        try {
+            /**
+             * Extract informations from the request. Based on those we can route the request.
+             *
+             * Filter first by HTTP method.
+             * After that, we can need to determine if we are in one of the situations:
+             * - collection - when we have no ID on the path
+             * - multiple elements - many ID's
+             * - individual element - we have one ID
+             * - relationship - we have ID and a relationship
+             * - field - we have ID and a field name
+             *
+             * No extra processing needs to be done - body parsing, etc.
+             */
+
+            switch (request.getMethod()) {
+                case GET:
+                    response = apiGet.handle(request);
+                    break;
+                case POST:
+                    response = apiPost.handle(request);
+                    break;
+                case PATCH:
+                    response = apiPatch.handle(request);
+                    break;
+                case DELETE:
+                    response = apiDelete.handle(request);
+                    break;
+                default:
+                    throw new MethodNotFoundException(request);
+            }
+
+        } catch (KatharsisMatchingException ke) {
+            response.setHttpStatus(406);
+        } catch (Exception e) {
+            response = toErrorResponse(e, 422);
```

This is is a bug. You're going to catch your own `MethodNotFoundException` and return a 422? 422 is a WebDav error and has no application at all. This should throw an `UnsupportedOperationException` and result in a 405: `METHOD_NOT_ALLOWED`

Also, what about `OPTIONS` and `HEAD` ?
If you look at the spec for for 405 `HEAD` should never report an error.